src/components/time-input/TimeInput.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import { defineMessages, injectIntl, FormattedMessage, WrappedComponentProps } from 'react-intl';
+import { TooltipPosition } from '../tooltip';
+import { parseTimeFromString } from './TimeInputUtils';
+// @ts-ignore flow import
+import TextInput from '../text-input';
+import ClockBadge16 from '../../icon/line/ClockBadge16';
+
+import './TimeInput.scss';
+
+const messages = defineMessages({
+    invalidTimeError: {
+        defaultMessage: 'Invalid time format. Enter a time in the format HH:MM A.',
+        description: 'Error message for invalid time formats. "HH:MM A" should be localized.',
+        id: 'boxui.timeInput.invalidTimeError',
+    },
+});
+export interface TimeInputProps extends WrappedComponentProps {
+    /** className - CSS class for the component */
+    className?: string;
+    /** errorTooltipPosition - Position for the error tooltip */
+    errorTooltipPosition?: TooltipPosition;
+    /** hideLabel - Whether the label should be hidden */
+    hideLabel?: boolean;
+    /** label - Label for the time input */
+    label?: React.ReactNode;
+    /** initialDate - Date object for initializing the time input */
+    initialDate?: Date;
+    /** isRequired - Whether the time input is required */
+    isRequired?: boolean;
+    /**
```

Comment:
So, you'd always want the value from `onBlur`, right? `onChange` isn't parsed. I wonder if that creates any edge cases. For example, in a `<form>`, if I press cmd-enter will the blur event be called?

Reply:
I also wonder if you want to do something to enable refs to work.

Reply:
Good call. I removed the `onChange` prop and added handlers for certain keys (escape, enter, command, control).